### PR TITLE
more predictable approach to closing the cookie banner

### PR DESCRIPTION
**Pull request by masone**
https://app.circleci.com/pipelines/github/carforyou/carforyou-ad-data-fetcher failing often. attempted fix by removing the dependency on the cookie. we moved away from this approach in `content-wider-than-screen`, which is much more stable since. maybe it helps here too. the path is now pretty predictable: 

1. always explicitly clear the cookies (should be the case anyway, but hey...)
2. as the banner is now always shown, we can simply always close the banner without checking for the cookie

maybe comes with a little bit of overhead that could have been saved with the previous approach where the cookie would persist between the test for each language

## How to test

```
npm run debug:prod
```

### CODE REVIEW

File: cypress/integration/fetchAds.js [hunk 1]
```diff
@@ -30,26 +33,21 @@ describe("Extract ads", () => {
             )
           })
         },
-      }).then(() => {
```
Comment (ufkaya) [inline cypress/integration/fetchAds.js:33]: Why did you remove it? Its not needed anymore?

**Review by ufkaya — APPROVED** [review]
I also tried to always accept cookies but I had some issues, so if you manage to fix the failing test, then I wont merge my [PR](https://github.com/carforyou/carforyou-ad-data-fetcher/pull/34l)

File: cypress/integration/fetchAds.js [hunk 2]
```diff
@@ -30,26 +33,21 @@ describe("Extract ads", () => {
...
-            cy.get("div .ad-loaded > #tatm-adHpEmotional[data-ad]", {
-              timeout: 20000,
-            }).then(($ad) => {
-              const ad =
-                $ad[0].dataset && $ad[0].dataset.ad
-                  ? JSON.parse($ad[0].dataset.ad)
-                  : {}
-              cy.writeFile(`${targetPath}${fileName}`, ad)
-            })
+      cy.get("#onetrust-banner-sdk")
```
Comment (GutierrezGo) [inline cypress/integration/fetchAds.js:38]: I think now it won't find this element. What happens if you leave the `then`, after visiting the page?

File: cypress/integration/fetchAds.js (hunk 1 above)
Comment (masone) [reply cypress/integration/fetchAds.js:33]: `cy` commands are ran synchronously. writing `cy.visit().then(() => { cy.visit() })` is exactly the same as `cy.visit(); cy.visit()`. we should only chain / use `then` when we are interested in the output of that command, for example when reading a value from a DOM node

one of the actions that made integration tests less flaky in `listings` was to remove all unnecessary chaining. less nesting, less complexity, less room for issues we guessed...

File: cypress/integration/fetchAds.js (hunk 2 above)
Comment (masone) [reply cypress/integration/fetchAds.js:38]: yeah, banner doesn't load in time randomly for one of the 4 languages... 💩

File: cypress/integration/fetchAds.js (hunk 1 above)
Comment (masone) [reply cypress/integration/fetchAds.js:33]: it does make a difference after all. I'm not sure, maybe it's because without the `then`, it's not properly waiting for our custom promise in `onBeforeLoad`? reverting back to the `then`... thanks for the pointer

**Comment by masone** [comment]
This is the best I can come up with. Giving it a shot... works relatively stable locally, but you know... CI can have quirks 🤞
